wp-includes/mysql/class-mysql-lexer.php

```diff
@@ -0,0 +1,2752 @@
+		self::SOURCE_USER_SYMBOL                     => 80000,
+		self::SOURCE_ZSTD_COMPRESSION_LEVEL_SYMBOL   => 80000,
+		self::SRID_SYMBOL                            => 80000,
+		self::STREAM_SYMBOL                          => 80019,
+		self::SYSTEM_SYMBOL                          => 80000,
+		self::THREAD_PRIORITY_SYMBOL                 => 80000,
+		self::TIES_SYMBOL                            => 80000,
+		self::TLS_SYMBOL                             => 80016,
+		self::UNBOUNDED_SYMBOL                       => 80000,
+		self::VCPU_SYMBOL                            => 80000,
+		self::VISIBLE_SYMBOL                         => 80000,
+		self::WINDOW_SYMBOL                          => 80000,
+		self::ZONE_SYMBOL                            => 80022,
+	);
+
+	/**
+	 * Identifier-like strings that may represent underscore-prefixed charset names.
+	 *
+	 * Includes charsets from both MySQL 5 and 8; via "SHOW CHARACTER SET"/docs:
+	 *   https://dev.mysql.com/doc/refman/5.7/en/charset-charsets.html
+	 *   https://dev.mysql.com/doc/refman/8.4/en/charset-charsets.html
+	 */
+	const UNDERSCORE_CHARSETS = array(
+		'_armscii8' => true,
+		'_ascii'    => true,
+		'_big5'     => true,
+		'_binary'   => true,
+		'_cp1250'   => true,
+		'_cp1251'   => true,
+		'_cp1256'   => true,
+		'_cp1257'   => true,
+		'_cp850'    => true,
+		'_cp852'    => true,
+		'_cp866'    => true,
+		'_cp932'    => true,
+		'_dec8'     => true,
+		'_eucjpms'  => true,
+		'_euckr'    => true,
+		'_gb18030'  => true,
+		'_gb2312'   => true,
+		'_gbk'      => true,
+		'_geostd8'  => true,
+		'_greek'    => true,
+		'_hebrew'   => true,
+		'_hp8'      => true,
+		'_keybcs2'  => true,
+		'_koi8r'    => true,
+		'_koi8u'    => true,
+		'_latin1'   => true,
+		'_latin2'   => true,
+		'_latin5'   => true,
+		'_latin7'   => true,
+		'_macce'    => true,
+		'_macroman' => true,
+		'_sjis'     => true,
+		'_swe7'     => true,
+		'_tis620'   => true,
+		'_ucs2'     => true,
+		'_ujis'     => true,
+		'_utf16'    => true,
+		'_utf16le'  => true,
+		'_utf32'    => true,
+		'_utf8'     => true,
+		'_utf8mb3'  => true,
+		'_utf8mb4'  => true,
+	);
+
+	protected $input;
+	protected $c; // Current character.
+	protected $n; // Next character.
+	protected $position = 0;
+	protected $token;
+	protected $text    = '';
+	protected $channel = self::CHANNEL_DEFAULT;
+	public $type;
+	protected $token_instance;
+	protected $server_version;
+	protected $sql_modes;
+	protected $in_version_comment = false;
+
+	public function __construct( string $input, int $server_version = 80038, int $sql_modes = 0 ) {
+		$this->input          = $input;
+		$this->server_version = $server_version;
+		$this->sql_modes      = $sql_modes;
+		$this->c              = $this->input[ $this->position ] ?? null;
+		$this->n              = $this->input[ $this->position + 1 ] ?? null;
+	}
+
+	public function is_sql_mode_active( int $mode ): bool {
+		return ( $this->sql_modes & $mode ) !== 0;
+	}
+
+	public function get_server_version() {
```

Comment:
I got confused at the beginning because there's no MySQL server involved here. How about `get_syntax_version`?

Reply:
@adamziel How about simply `get_mysql_version()`? Not sure if the word "syntax" may confuse someone here.

Reply:
Sounds good! It gives a slight impression that we're actually connecting to the mysql server somehow, but the alternative such as `get_mysql_syntax_version` sparks questions like "is the syntax version different from the database version?"

So yes, let's go with `get_mysql_version`